CI staging namespace

bbguimaraes opened the pull request:
Secret mirroring is still missing.  For intended usage, see:

[8c3560f](https://github.com/openshift/release/commit/8c3560ff25b16516b2d85d017018824aa2df0227)

petr-muller commented on hack/mkpjpod.sh:
```diff
@@ -0,0 +1,28 @@
+fi
+job=$1
+img=registry.svc.ci.openshift.org/ci/test-infra:binaries
+podman run --rm -iv "$PWD:/tmp/release:Z" -w /tmp/release "$img" bash <<-EOF
```
Hm are we that confident everyone is running `podman` already?

stevekuznetsov replied:
I'm not :)

stevekuznetsov replied:
Also the `:Z` should not be necessary, `:z` should suffice

bbguimaraes replied:
> Hm are we that confident everyone is running podman already?

Can we make them? =)

> I'm not :)

Shame on you.

> Also the :Z should not be necessary, :z should suffice

I always think of it as `z` == "security", `Z` == "more security". I'll have to find a better mnemonic (or RTFM).

stevekuznetsov commented on cluster/ci/config/prow/plugins.yaml:
```diff
@@ -403,55 +403,130 @@ config_updater:
 
     cluster/test-deploy/gcp/*.yaml:
       name: cluster-profile-gcp
+    cluster/test-deploy/gcp/*.yaml:
```
Maybe we should change the upload plugin to allow a list of namespaces?

bbguimaraes commented:
Updated:

- rebased
- added the `ci-admins` group as admins in the namespace
- added secret mirroring configuration

I have deployed the secret mirroring controller and everything in this PR on `api.ci` to test the deployment configuration. Namespace `ci-stg` is live and secrets are being mirrored. I have left out:

1. `ci-operator` service account
2. configmaps
3. `config_updater` configuration
3. rolebinding that gives regular users access

I will do (1) and apply (2) manually tomorrow and do a full test, then maybe tackle the `config_updater` configuration change.

stevekuznetsov commented on cluster/ci/config/prow/openshift/ci-operator/stage.yaml:
```diff
@@ -0,0 +1,28 @@
+  apiVersion: authorization.openshift.io/v1
+  metadata:
+    name: ci-stg-user
+  rules:
```
Do we need create on events here, too?

bbguimaraes replied:
`ci-operator` will use its own service account. This role is for the individual user accounts, which should interface with the namespace exclusively by creating pods generated by `mkpod`.

The reason ConfigMaps are included is to allow template authors (or adventurous users) to create their own template CM and change the job to use it (which have to has to be in the same namespace per kube's restrictions).

stevekuznetsov replied:
Of course, that makes sense
